cli/src/main.rs

```diff
@@ -0,0 +1,262 @@
+use bson::{doc, Document};
+use clap::Parser;
+use mongodb::sync::{Client, Collection};
+use mongosql::{catalog::Catalog, json_schema::Schema, Namespace};
+use std::collections::{BTreeMap, BTreeSet};
+
+const SQL_SCHEMAS_COLLECTION: &str = "__sql_schemas";
+
+#[derive(Debug)]
+struct CliError(String);
+
+impl std::fmt::Display for CliError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl<T> From<T> for CliError
+where
+    T: std::error::Error,
+{
+    fn from(e: T) -> Self {
+        CliError(e.to_string())
+    }
+}
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about=None)]
+struct Cli {
+    #[arg(
+        short,
+        long,
+        help = "The current database where collections in the query are assumed to live (cross database queries are not supported), default = test"
+    )]
+    db: Option<String>,
+    #[arg(index = 1, help = "The query to translate")]
+    query: String,
+    #[arg(
+        short,
+        long,
+        help = "translation, automatically true if result not set"
+    )]
+    translation: bool,
+    #[arg(
+        short,
+        long,
+        help = "Display the result of this query when applied against the mongodb uri"
+    )]
+    result: bool,
+    #[arg(
+        short,
+        long,
+        help = "The mongodb uri, default = mongodb://localhost:27017"
+    )]
+    uri: Option<String>,
+}
+
+fn main() -> Result<(), CliError> {
+    let args = Cli::parse();
+
+    let uri = args.uri.unwrap_or("mongodb://localhost:27017".to_string());
+    let current_db = args.db.unwrap_or("test".to_string());
+    let query = args.query;
+    let namespaces = mongosql::get_namespaces(current_db.as_str(), query.as_str())?;
+    let catalog = get_schema_catalog(uri.as_str(), current_db.as_str(), namespaces)?;
+    let translation = mongosql::translate_sql(
+        current_db.as_str(),
+        query.as_str(),
+        &catalog,
+        mongosql::options::SqlOptions::default(),
+    )?;
+    let print_translation = |pipeline| {
+        println!(
+            "target_db: {},\ntarget_collection: {:?},\nschema:{:?},\npipeline:",
+            translation.target_db, translation.target_collection, translation.result_set_schema
+        );
```

Comment:
Try this change, it prints out the schema in a nice readable way.  The pipeline could use a pretty print option if you want, although it is also good the way it is since it's printing a newline for each entry in the pipeline. 
```suggestion
    let print_translation = |pipeline: bson::Bson| -> Result<(), CliError> {
        let schema = serde_json::to_string_pretty(&translation.result_set_schema)
            .map_err(|e| CliError(e.to_string()))?;
        println!(
            "target_db: {},\ntarget_collection: {:?},\nschema:\n{}\npipeline:",
            translation.target_db, translation.target_collection, schema
        );
```